### ISSUE TITLE
[client] Accessibility: Identify the current annotation

### ISSUE THREAD

**Issue by chrisshaw** [hypothesis/product-backlog#1341]
**Improve annotation navigation by uniquely identifying each annotation.**

From our WCAG 2.1 Audit. This was also identified by a student user in a accessibility call.

> ### Labels for article elements and related controls
> Currently, each annotation uses an `article` element, which can improve navigation for assistive technology users. However, each article element does not have a unique label. As a result, when interacting with the annotations through the list of landmarks all landmarks are announced the same (“article”).
> 
> ![image](https://user-images.githubusercontent.com/2966918/165677127-224c5467-dada-40d8-826d-1ea2ada86aac.png)
>
> _Voiceover landmarks menu showing a list of five article elements with no additional labels._
> 
> To improve the accessibility and ease navigation between annotations, consider:
> 
> - Adding an aria-labelledby attribute and linking it to the article heading (the heading that indicates who posted the annotation)
> - Adding an aria-describedby attribute to the article and associate with the highlighted portion of the annotation.
> - Adding an aria-describedby attribute to the edit, reply, and delete buttons and associate them with the annotation as well.
> 
> ### Resources for article regions
> [MDN Article documentation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/article_role&sa=D&source=editors&ust=1643394101527357&usg=AOvVaw3Zvpmo1hcLZf4oLv9USAMa)
> 
> While the following reference a feed pattern which does not exactly match the list of annotations, the example and related documentation are great examples of using aria-labelledby and aria-describedby for <article> elements:
> 
> [W3C Feed example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/feed/feedDisplay.html&sa=D&source=editors&ust=1643394101527895&usg=AOvVaw0ffa-B4_rTL5la8Up6q9rG)
> [W3C Feed documentation](https://www.w3.org/TR/wai-aria-practices-1.1/examples/feed/feed.html&sa=D&source=editors&ust=1643394101528116&usg=AOvVaw15cxaEIUjjc-frIzzmknia)

**Comment by chrisshaw** [comment]
**Issue was linked to Harvestr Discovery:**  [Know what annotation I'm modifying via screen readers](https://app.harvestr.io/components/0/list/2ccn8e85P)

**Comment by lyzadanger** [comment]
Reviewed: Need to add `aria-` attribute(s) to `Annotation` `article` element.

**Comment by lyzadanger** [comment]
I appreciate the level of detail in the issue description; it has helped me find good resources to figure out good approaches to this.

The requestor has suggested using the author's name as a "label" and the "highlighted text" (in some of the front-end code, we call this a "quote", which is a useful term, as "highlight" is overloaded) to "describe" the annotation `article` element. I used this as a starting point and then considered all of the "flavors" of annotations to come up with a proposed recommendation as follows.

For reference, the `aria-*` attributes in play here are:

* `aria-label`: Takes a string value to label the element. There's also `aria-labelledby`, which takes one or more element IDs that contain labelling for the element. In the cases here, I'm proposing cooking up a label string, but we have `aria-labelledby` in our back pocket should we need it.
* `aria-describedby`: Takes one or more element IDs that describe the element.

### Proposed `aria` descriptive attributes for annotation `<article>` elements

* Top-level annotations
	* `aria-label`:  `“Annotation by <user>”`
	* `aria-describedby`: `<annotation quote element id>`
* Reply annotations
	*  `aria-label`: `“Reply by <user>”`
* Top-level missing (deleted) annotations
	* `aria-label`:  `“Annotation with unavailable content”`
* Missing reply annotation
	* `aria-label`: `“Reply with unavailable content”`
* Highlight (always top-level)
	* `aria-label`: `“Highlight by <user>”`
	* `aria-describedby`: `<annotation quote element id>`


(Updated to simplify `describedby` approach).

**Comment by lyzadanger** [comment]
Given the original intent of this issue:

> Improve annotation navigation by uniquely identifying each annotation.

I believe we have satisfied this by providing a contextual `aria-label` to each annotation `article` element. That context should apply to the entire annotation.
